src/resmom/linux/mom_mach.c

```diff
@@ -5599,17 +5591,43 @@ mom_get_sample(void)
 		}
 
 		ps = &proc_info[nproc];
-		if (fscanf(fd, stat_str, &ps->pid, path, &ps->state,
-			&ps->ppid, &ps->pgrp, &ps->session,
-			&ps->flags, &ps->utime, &ps->stime,
-			&ps->cutime, &ps->cstime, &jiffies,
-			&ps->vsize, &ps->rss) != 14) {
+		stat_str = choose_procflagsfmt();
+		if (stat_str == NULL) {
+			log_err(errno, __func__, "choose_procflagsfmt allocation failed");
+			return PBSE_INTERNAL;
+		}
+		if (fscanf(fd, stat_str,
+			   &ps->pid,		/* "%d "	1  pid %d The process id */
+			   path,		/* "(%[^)]) "	2  comm %s The filename of the executable */
+			   &ps->state,		/* "%c "	3  state %c "RSDZTW" */
+			   &ps->ppid,		/* "%d "	4  ppid %d The PID of the parent */
+			   &ps->pgrp,		/* "%d "	5  pgrp %d The process group ID */
+			   &ps->session,	/* "%d "	6  session %d The session ID */
+				   		/* "%*d "	7  ignored:  tty_nr */
+	 		   			/* "%*d "	8  ignored:  tpgid */
+			   &ps->flags,		/* "%u or %lu"	9  flags */
+					   	/* "%*lu "	10 ignored:  minflt */
+					   	/* "%*lu "	11 ignored:  cminflt */
+					   	/* "%*lu "	12 ignored:  majflt */
+					   	/* "%*lu "	13 ignored:  cmajflt */
+			   &ps->utime,		/* "%lu "	14 utime %lu */
+			   &ps->stime,		/* "%lu "	15 stime %lu */
+			   &ps->cutime,		/* "%ld "	16 cutime %ld */
+			   &ps->cstime,		/* "%ld "	17 cstime %ld */
+				   		/* "%*ld "	18 ignored:  priority %ld */
+			   			/* "%*ld "	19 ignored:  nice %ld */
+			   			/* "%*ld "	20 ignored:  num_threads %ld */
+			   			/* "%*ld "	21 ignored:  itrealvalue %ld - no longer maintained */
+			   &starttime,		/* "%llu "	22 starttime (was %lu before Linux 2.6 - see proc(5) for conversion details */
+			   &ps->vsize,		/* "%lu "	23 vsize (bytes) */
+			   &ps->rss		/* "%ld "	24 rss (number of pages) */
+			) != 14) {
```

Comment:
Maybe I'm just being dense here, but don't you need a variable to correspond with every component of your format string even if some elements will end up being ignored? The format string has 24 members as it is currently defined.